### ISSUE TITLE
FR: Wait for string/regex to show up in server output

### ISSUE THREAD

**Issue by anishkny** [bahmutov/start-server-and-test#316]
**Feature Request**

Allow waiting on some string/regex to appear in the server output text (stdout/stderr) instead of port/URL to be used.

Some servers may not allow configuring or determining which port they use, or may not even use ports. Seems like the only way to determine if such a server has started is via some text appearing on stdout/stderr.

For example when using `stripe-cli` to listen to webhooks locally for testing, we only get the output:
```
⣟ Getting ready... > Ready! Your webhook signing secret is whsec_xxxxxxxxxxxx (^C to quit)
```

It would be nice to have something like:

```jsonc
{
    "scripts": {
        "start": "stripe listen ...",
        "test": "./run-tests.sh",

        // Wait for regex /Ready!/ to appear in output of "npm start":
        "ci": "start-server-and-test start /Ready!/ test" 
    }
}
```

See related https://github.com/stripe/stripe-cli/discussions/756.
